rocky/reports/report_types/vulnerability_report/report.py

```diff
@@ -0,0 +1,149 @@
+from collections import Counter
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict, List, Union
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.findings import Finding, KATFindingType, RiskLevelSeverity
+from octopoes.models.ooi.network import IPAddressV4, IPAddressV6
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+
+class VulnerabilityReport(Report):
+    id = "vulnerability-report"
+    name = _("Vulnerability Report")
+    description: str = _("Vulnerabilities found are grouped for each system.")
+    plugins = {
+        "required": ["dns-records", "nmap", "website-software"],
+        "optional": ["nmap-udp", "nmap-ports", "shodan"],
+    }
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    template_path = "vulnerability_report/report.html"
+
+    def get_finding_valid_time_history(self, reference: str) -> List[datetime]:
+        transaction_record = self.octopoes_api_connector.get_history(reference=reference)
```

Comment:
Can we not be more specific?
https://github.com/minvws/nl-kat-coordination/blob/727b71ae75153b0ebf621f5e43f61688f5137030/octopoes/octopoes/core/service.py#L107
Order by date, and limit to 1 for the oldest?
The most recent should be on our currently loaded OOI anyway.

Also, I'm not sure we need the Docs, as we are not accessing them, so that would trim down on the data being returned fomr XTDB.

Reply:
This is about the finding. Findings are only created once, by the bit

Reply:
Should be changed to get info of ooi

Reply:
Why do we want info of OOI? This only gives transaction records datetime info of oldest and newest.

Reply:
Because the first and last seen of the ooi is far more informational than the first and last seen of the finding. This is because the first seen of the ooi always equals the first seen of the finding (currently), and the last seen actually shows the last wappalyzer scan while the last seen of the finding doenst exist since the bit that created the finding only ran once.